### PR TITLE
Print console log

### PR DESCRIPTION
**Pull request by XiongKezhi**
Use TaskListener to print console log

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java [hunk 1]
```diff
@@ -54,10 +54,10 @@ public void publish(final ChecksDetails details) {
                     credentials);
             GHCheckRunBuilder builder = createBuilder(gitHub, new GitHubChecksDetails(details));
             builder.create();
+            listener.getLogger().println("GitHub checks have been published.");
         }
         catch (IllegalStateException | IOException e) {
-            //TODO: log to the build console
-            LOGGER.log(Level.WARN, "Could not publish GitHub check run", e);
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java:60]: still worth logging this to the system?

File: src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherFactory.java [hunk 2]
```diff
@@ -35,28 +35,32 @@ public GitHubChecksPublisherFactory() {
...
+        return createPublisher(new GitHubChecksContext(job), listener);
     }
 
-    protected Optional<ChecksPublisher> createPublisher(final GitHubChecksContext context) {
+    protected Optional<ChecksPublisher> createPublisher(final GitHubChecksContext context,
+                                                        final TaskListener listener) {
         Job<?, ?> job = context.getJob();
         Optional<GitHubSCMSource> source = scmFacade.findGitHubSCMSource(job);
         if (!source.isPresent()) {
+            listener.getLogger().println("Failed creating GitHub checks publisher: no GitHub SCM source found.");
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherFactory.java:52]:
```suggestion
            listener.getLogger().println("Skipped publishing GitHub checks: no GitHub SCM found.");
```

File: src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherFactory.java [hunk 3]
```diff
@@ -35,28 +35,32 @@ public GitHubChecksPublisherFactory() {
...
             return Optional.empty();
         }
 
         String credentialsId = source.get().getCredentialsId();
         if (credentialsId == null
                 || !scmFacade.findGitHubAppCredentials(job, credentialsId).isPresent()) {
+            listener.getLogger().println("Failed creating GitHub checks publisher: no GitHub APP credentials found.");
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherFactory.java:59]:
```suggestion
            listener.getLogger().println("Skipped publishing GitHub checks: no GitHub APP credentials found, see https://github.com/jenkinsci/github-branch-source-plugin/blob/master/docs/github-app.adoc");
```

File: src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherFactory.java [hunk 4]
```diff
@@ -35,28 +35,32 @@ public GitHubChecksPublisherFactory() {
...
             return Optional.empty();
         }
 
-        return Optional.of(new GitHubChecksPublisher(context));
+        listener.getLogger().println("Using GitHub checks publisher.");
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherFactory.java:63]:
```suggestion
        listener.getLogger().println("Publishing GitHub checks.");
```

**Comment by XiongKezhi** [comment]
irrelevant CI failure

File: src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java [hunk 5]
```diff
@@ -54,10 +57,12 @@ public void publish(final ChecksDetails details) {
...
+            String message = "Failed Publishing GitHub checks: " + e;
+            LOGGER.log(Level.WARN, message);
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java:64]: always log the exception with the throwable arg
```suggestion
            LOGGER.log(Level.WARN, message, e);
```

File: src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java [hunk 6]
```diff
@@ -54,10 +57,12 @@ public void publish(final ChecksDetails details) {
...
+            listener.getLogger().println(message);
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java:65]:
```suggestion
            listener.getLogger().println(message + e);
```

File: src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java [hunk 7]
```diff
@@ -54,10 +57,12 @@ public void publish(final ChecksDetails details) {
                     credentials);
             GHCheckRunBuilder builder = createBuilder(gitHub, new GitHubChecksDetails(details));
             builder.create();
+            listener.getLogger().println("GitHub checks have been published.");
         }
         catch (IllegalStateException | IOException e) {
-            //TODO: log to the build console
-            LOGGER.log(Level.WARN, "Could not publish GitHub check run", e);
+            String message = "Failed Publishing GitHub checks: " + e;
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java:63]:
```suggestion
            String message = "Failed Publishing GitHub checks: ";
```

**Pull request merged**